compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -867,6 +867,16 @@ impl<'tcx> Printer<'tcx> for SymbolMangler<'tcx> {
             // are effectively living in their parent modules.
             DefPathData::ForeignMod => return print_prefix(self),
 
+            // Global asm are handled similar to shims.
+            DefPathData::GlobalAsm => {
```

Comment:
Why not just give this a prefix? Then we'll encode it with that prefix + disambiguator + empty name.